### DISCUSSION TITLE
HTML error

### DISCUSSION THREAD

**Discussion by farmboy-dev** [grafana/grafana-infinity-datasource#582] [Q&A]
Hi

Thank you for the great plugin.

I'm trying to parse a html page.

but I keep getting this message "Cannot read properties of undefined (reading 'DOCUMENT_MODE')"

My seeting is type: HTML, Parser: Default, Source: URL Format: Table Method:GET, URL: my_URL

How to figure it out?

**Comment by gbmd** [comment]
I second this. Updated Grafana 9 to 9.5.1 and infinity to 1.4.1, now all my html request panels display this error. (JSON works)

**Reply by gbmd** [reply]
Here's what my query inspector states: 

`{
  "request": {
    "url": "api/ds/query",
    "method": "POST",
    "data": {
      "queries": [
        {
          "datasource": {
            "type": "yesoreyeram-infinity-datasource",
            "uid": "u7q0Ovwnz"
          },
          "columns": [
            {
              "selector": "h3",
              "text": "title",
              "type": "string"
            },
            {
              "selector": "span.meta",
              "text": "time",
              "type": "string"
            }
          ],
          "filters": [],
          "format": "table",
          "global_query_id": "",
          "refId": "A",
          "root_selector": "article",
          "source": "url",
          "type": "html",
          "url": "https://www.electrive.net/",
          "url_options": {
            "data": "",
            "method": "GET",
            "body_type": "",
            "body_content_type": "",
            "body_graphql_query": ""
          },
          "parser": "simple",
          "datasourceId": 11,
          "intervalMs": 10000,
          "maxDataPoints": 2371
        }
      ],
      "range": {
        "from": "2023-05-02T02:42:34.150Z",
        "to": "2023-05-02T08:42:34.150Z",
        "raw": {
          "from": "now-6h",
          "to": "now"
        }
      },
      "from": "1682995354150",
      "to": "1683016954150"
    },
    "hideFromInspector": false
  },
  "response": {
    "results": {
      "A": {
        "status": 200,
        "frames": [
          {
            "schema": {
              "name": "A",
              "refId": "A",
              "meta": {
                "typeVersion": [
                  0,
                  0
                ],
                "custom": {
                  "data": "*****(Valid HTML of the website is here - cut it for simplicity)******",
                  "duration": 29642177,
                  "error": "",
                  "query": {
                    "alias": "",
                    "columns": [
                      {
                        "selector": "h3",
                        "text": "title",
                        "timestampFormat": "",
                        "type": "string"
                      },
                      {
                        "selector": "span.meta",
                        "text": "time",
                        "timestampFormat": "",
                        "type": "string"
                      }
                    ],
                    "computed_columns": [],
                    "csv_options": {
                      "columns": "",
                      "comment": "",
                      "delimiter": "",
                      "relax_column_count": false,
                      "skip_empty_lines": false,
                      "skip_lines_with_error": false
                    },
                    "data": "",
                    "dataOverrides": null,
                    "expression": "",
                    "filterExpression": "",
                    "filters": [],
                    "format": "table",
                    "global_query_id": "",
                    "groq": "",
                    "json_options": {
                      "columnar": false,
                      "root_is_not_array": false
                    },
                    "parser": "simple",
                    "query_mode": "",
                    "refId": "A",
                    "root_selector": "article",
                    "seriesCount": 0,
                    "source": "url",
                    "sqlite_query": "",
                    "summarizeBy": "",
                    "summarizeExpression": "",
                    "type": "html",
                    "uql": "",
                    "url": "https://www.electrive.net/",
                    "url_options": {
                      "body_content_type": "",
                      "body_form": null,
                      "body_graphql_query": "",
                      "body_type": "",
                      "data": "",
                      "headers": null,
                      "method": "GET",
                      "params": null
                    }
                  },
                  "responseCodeFromServer": 200
                },
                "executedQueryString": "###############\n## URL\n###############\n\nhttps://www.electrive.net/\n\n###############\n## Curl Command\n###############\n\ncurl -X 'GET' 'https://www.electrive.net/'"
              },
              "fields": []
            },
            "data": {
              "values": []
            }
          }
        ],
        "refId": "A"
      }
    }
  }
}`

**Reply by gbmd** [reply]
Downgrade to 1.2.0 fixed it - will stay there for now

**Comment by yesoreyeram** [comment]
Have you tried allowed hosts config in the data source configuration?

**Reply by diegotsm1** [reply]
Mine has the same problem.
Yes, I previously tried to configure the allowed host in the data source configuration. But then I removed the configuration and it keeps giving the same error.
 @yesoreyeram Can you help me?
